packages/db-core/index.js

```diff
@@ -34,4 +34,46 @@ module.exports = fp(async function (app, opts) {
   }
 })
 
+function shouldRegister (opts, configKey) {
```

Comment:
I probably went a little verbose with the logic and comments but wanted to get it straight in my head. I can consolidate if it's too much.